conversationcallback-impl/src/main/java/ai/labs/callback/impl/ConversationCallbackTask.java

```diff
@@ -0,0 +1,114 @@
+package ai.labs.callback.impl;
+
+import ai.labs.callback.IConversationCallback;
+import ai.labs.callback.http.ConversationCallback;
+import ai.labs.callback.model.ConversationDataRequest;
+import ai.labs.callback.model.ConversationDataResponse;
+import ai.labs.lifecycle.AbstractLifecycleTask;
+import ai.labs.lifecycle.ILifecycleTask;
+import ai.labs.lifecycle.LifecycleException;
+import ai.labs.lifecycle.PackageConfigurationException;
+import ai.labs.memory.ConversationMemoryUtilities;
+import ai.labs.memory.IConversationMemory;
+import ai.labs.memory.model.ConversationMemorySnapshot;
+import ai.labs.persistence.IResourceStore;
+
+import javax.inject.Inject;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Created by rpi on 08.02.2017.
+ */
+public class ConversationCallbackTask extends AbstractLifecycleTask implements ILifecycleTask {
+    private static final String KEY_CALLBACK_URI = "callbackUri";
+    private static final String KEY_TIMEOUT_IN_MILLIS = "timeoutInMillis";
+
+    private final IConversationCallback conversationCallback;
+    private URI callback;
+    private long timeoutInMillis;
+
+    @Inject
+    public ConversationCallbackTask(IConversationCallback conversationCallback) {
+        this.conversationCallback = conversationCallback;
+    }
+
+    @Override
+    public String getId() {
+        return ConversationCallback.class.toString();
+    }
+
+    @Override
+    public Object getComponent() {
+        return conversationCallback;
+    }
+
+    @Override
+    public List<String> getComponentDependencies() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public List<String> getOutputDependencies() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void executeTask(IConversationMemory memory) throws LifecycleException {
+        try {
+            ConversationDataRequest request = new ConversationDataRequest();
+            request.setConversationMemorySnapshot(ConversationMemoryUtilities.convertConversationMemory(memory));
+            ConversationDataResponse response =
+                    conversationCallback.doExternalCall(callback, request, timeoutInMillis);
+
+            if (response.getHttpCode() == 200) {
```

Comment:
Is it always right to check for HTTP status 200 here? An external call might as well return some different success status depending on what action it is performing.